features/telemetry/protocols/sflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/sflow/parser/SampleDatagramEnricher.java

```diff
@@ -53,55 +54,64 @@
 public class SampleDatagramEnricher {
 
     private final DnsResolver dnsResolver;
+    private boolean dnsLookupsEnabled;
 
-    public SampleDatagramEnricher(DnsResolver dnsResolver) {
+    public SampleDatagramEnricher(DnsResolver dnsResolver, boolean dnsLookupsEnabled) {
         this.dnsResolver = Objects.requireNonNull(dnsResolver);
+        this.dnsLookupsEnabled = dnsLookupsEnabled;
     }
 
     public CompletableFuture<SampleDatagramEnrichment> enrich(SampleDatagram datagram) {
-        final Set<InetAddress> addressesToReverseLookup = new HashSet<>();
-        datagram.visit(new SampleDatagramVisitor() {
-            @Override
-            public void accept(Inet4Header inet4Header) {
-                addressesToReverseLookup.add(inet4Header.getSrcAddress());
-                addressesToReverseLookup.add(inet4Header.getDstAddress());
-            }
-
-            @Override
-            public void accept(Inet6Header inet6Header) {
-                addressesToReverseLookup.add(inet6Header.getSrcAddress());
-                addressesToReverseLookup.add(inet6Header.getDstAddress());
-            }
-
-            @Override
-            public void accept(IpV4 ipV4) {
-                addressesToReverseLookup.add(ipV4.getAddress());
-            }
-
-            @Override
-            public void accept(IpV6 ipV6) {
-                addressesToReverseLookup.add(ipV6.getAddress());
-            }
-        });
-
-        final Map<InetAddress, String> hostnamesByAddress = new HashMap<>(addressesToReverseLookup.size());
-        final List<CompletableFuture<Optional<String>>> reverseLookupFutures = addressesToReverseLookup.stream()
-                .map(addr -> dnsResolver.reverseLookup(addr).whenComplete((hostname, ex) -> {
-                    if (ex != null) {
-                        synchronized (hostnamesByAddress) {
-                            hostnamesByAddress.put(addr, hostname.orElse(null));
+        if (this.dnsLookupsEnabled) {
```

Comment:
I'd change this to:
```suggestion
        if (!this.dnsLookupsEnabled) {
            final CompletableFuture<SampleDatagramEnrichment> emptyFuture = new CompletableFuture<>();
            final SampleDatagramEnrichment emptyEnrichment = new DefaultSampleDatagramEnrichment(Collections.<InetAddress, String>emptyMap());
            emptyFuture.complete(emptyEnrichment);
            return emptyFuture;
        }
```

this way we can avoid indenting the large block that actually does the work.

Reply:
Done